libkv.go

```diff
@@ -30,29 +33,98 @@ func NewLibKV(url *url.URL) *LibKV {
 
 	sourceType = ""
 	client = ""
+	options := store.Config{}
 
 	if url.Scheme == "consul" {
 		consul.Register()
 		sourceType = store.CONSUL
 		client = env.Getenv("CONSUL_HTTP_ADDR", "localhost:8500")
+		if timeout := env.Getenv("CONSUL_TIMEOUT", ""); timeout != "" {
+			num, err := strconv.ParseInt(timeout, 10, 16)
+			if err != nil {
+				logFatal("consul timeout error", err)
+			}
+			options.ConnectionTimeout = time.Duration(num) * time.Second
+		}
+		if ssl := env.Getenv("CONSUL_HTTP_SSL", ""); ssl != "" {
```

Comment:
Why is this necessary? Wouldn't the `CONSUL_HTTP_ADDR`'s scheme reflect whether it's HTTPS or not?

I do see that it doesn't have a scheme in the default above, and I think that should probably be changed.

Additionally, what might make sense (maybe in a followup PR though?) is to also support a `consul://` URL that has the host in it, something like:

```console
$ gomplate -d consul=consul+http://consul.server:8500/ -i '{{ ds "consul" "mykey" }}'
some value from consul...
```

Or with HTTPS:
```console
$ gomplate -d consul=consul+https://tls-protected-consul.server:8533/ -i '{{ ds "consul" "mykey" }}'
some value from consul...
```